src/utils/FetcherConfiguration.ts

```diff
@@ -0,0 +1,47 @@
+export enum FetcherResponseType {
+  JSON,
+  Buffer
+}
+
+export type RequestOptions = {
+  url: string
+  method: string
+  responseType: FetcherResponseType
+  // Time format accepted by ms: Examples: '0.5s', '2m', '3h', '100' (assumed to be milliseconds)
+  attempts: number // Number of attempts to perform the request
+  timeout: string // Time to abort the request. Time format accepted by ms
+  waitTime: string // Time to wait between attempts. Time format accepted by ms
+  body?: FormData | string
+  headers?: Record<string, string>
+}
+
+export class FetcherConfiguration {
```

Comment:
I think there is no need to have a class with static variables. They could just be exported `const`s.